File: sol/solution/0101-0200/0110/Solution.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int max(int a, int b) { return a > b ? a : b; }
int height(struct TreeNode* root) {
    if (!root) return 0;
    int left = height(root->left);
    if (left == -1) return -1;
    int right = height(root->right);
    if (right == -1) return -1;
    if (abs(left - right) > 1) return -1;
    return max(left, right) + 1;
}
bool isBalanced(struct TreeNode* root) {
    return height(root) != -1;
}

File: sol/solution/0101-0200/0110/Solution.ex
# Definition for a binary tree node.
#
# defmodule TreeNode do
#   @type t :: %__MODULE__{
#           val: integer,
#           left: TreeNode.t() | nil,
#           right: TreeNode.t() | nil
#         }
#   defstruct val: 0, left: nil, right: nil
# end
defmodule Solution do
  def is_balanced(nil), do: true
  def is_balanced(root), do: check(root) != -1
  defp check(nil), do: 0
  defp check(%TreeNode{left: left, right: right}) do
    l = check(left)
    if l == -1 do
      -1
    else
      r = check(right)
      if r == -1 or abs(l - r) > 1 do
        -1
      else
        max(l, r) + 1
      end
    end
  end
end
